Move static from_ivalue/to_ivalue to new shim_common.cpp

mikaylagawarecki opened the pull request:
Move `from_ivalue` and `to_ivalue` and their dependents `StableIValueBoxedKernel`, `aoti_torch_library_impl` `aoti_torch_call_dispatcher` into new (non-aoti shim_common.cpp)

This is in prep for the above PRs where I add v2s (`torch_call_dispatcher` and `torch_library_impl`) that are versioning aware

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #165285
* #165284
* #165283
* #164332
* #163683
* __->__ #166373
* #164356

mikaylagawarecki committed:
Move static from_ivalue/to_ivalue to new shim_common.cpp

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Move static from_ivalue/to_ivalue to new shim_common.cpp"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Move static from_ivalue/to_ivalue to new shim_common.cpp"




[ghstack-poisoned]

janeyx99 reviewed (commented):
looks fine to me, barring two questions

janeyx99 commented on build_variables.bzl:
```diff
@@ -482,6 +482,7 @@ inductor_core_resources = [
     "torch/csrc/inductor/aoti_torch/oss_proxy_executor.cpp",
     "torch/csrc/inductor/inductor_ops.cpp",
     "torch/csrc/jit/serialization/pickle.cpp",
+    "torch/csrc/shim_common.cpp",
```
hmmmmmmmmmmmmmmmmmmmmmmmm.

i think it may be important to figure out why that is 😛  otherwise this is a bit weird why it needs to be in inductor.

mikaylagawarecki replied:
I think the reason why is that this file includes the aoti shim (which is only in the inductor sources)

So if we put this in core_sources_common, it would be included by the mobile builds, but then the aoti shim would be excluded

janeyx99 replied:
Still not sure I fully understand if this is the right move, but given that it seems ok + that this is undoable, I am approving.

mikaylagawarecki committed:
Update on "Move static from_ivalue/to_ivalue to new shim_common.cpp"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Move static from_ivalue/to_ivalue to new shim_common.cpp"

Move `from_ivalue` and `to_ivalue` and their dependents `StableIValueBoxedKernel`, `aoti_torch_library_impl` `aoti_torch_call_dispatcher` into new (non-aoti shim_common.cpp)

This is in prep for the above PRs where I add v2s (`torch_call_dispatcher` and `torch_library_impl`) that are versioning aware




[ghstack-poisoned]
